readthedocs/doc_builder/backends/mkdocs.py

```diff
@@ -108,20 +108,22 @@ def append_conf(self, **__):
         docs_path = os.path.join(self.root_path, docs_dir)
 
         # RTD javascript writing
-        rtd_data = self.generate_rtd_data(docs_dir=docs_dir)
+        rtd_data = self.generate_rtd_data(docs_dir=docs_dir, mkdocs_config=user_config)
         with open(os.path.join(docs_path, 'readthedocs-data.js'), 'w') as f:
             f.write(rtd_data)
 
-    def generate_rtd_data(self, docs_dir):
+    def generate_rtd_data(self, docs_dir, mkdocs_config):
         """Generate template properties and render readthedocs-data.js."""
+        theme_name = mkdocs_config.get('theme_dir', 'readthedocs').rsplit('/')[-1]
```

Comment:
Since we are managing a path I think it's better to use `os.path`. Depending on the value of `theme_dir` this could fail:

```python
>>> '/home/humitos/docs/theme'.rsplit('/')
['', 'home', 'humitos', 'docs', 'theme']
>>> '/home/humitos/docs/theme/'.rsplit('/')
['', 'home', 'humitos', 'docs', 'theme', '']
>>> 
```

in the second case we will endup with an empty string.

I suggest something like this:

```python
>>> os.path.basename('/home/humitos/docs/theme/'.rstrip('/'))
'theme'
>>> os.path.basename('/home/humitos/docs/theme'.rstrip('/'))
'theme'
```

(you can achieve the same behaviour with your implementation just adding the `rstrip` --just in case)

Reply:
I hesitate to use `os.path` because I suspect MkDocs always uses `/` regardless of what `os.sep` is.

However, your point on `rstrip` is well taken. I'll update.